polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -1478,17 +1474,28 @@ ValueCategory MLIRScanner::VisitCastExpr(CastExpr *E) {
     return Visit(E->getSubExpr());
 
   case clang::CastKind::CK_AddressSpaceConversion: {
-    auto Scalar = Visit(E->getSubExpr());
-    // JLE_QUEL::TODO (II-201)
-    // assert(scalar.isReference);
-    auto PostTy = ReturnVal.getType().cast<MemRefType>().getElementType();
```

Comment:
`ReturnVal` is NULL for the issue being fixed, we should not rely on it at all and instead the fix computes the target address space from the cast expression. I have added an assert in DEBUG mode just to verify that, when ReturnVal *is* present the new code yields the same address space as the old one.